File: 3838-weighted-word-mapping/3838-weighted-word-mapping.cpp
class Solution {
public:
    string mapWordWeights(vector<string>& words, vector<int>& weights) {
        string result;
        for (const string& word : words) {
            int sumWeight = 0;
            for (char ch : word) {
                sumWeight += weights[ch - 'a'];
            }
            int value = sumWeight % 26;   
            result.push_back(char('z' - value));
        }
        return result;
    }
};

File: 3838-weighted-word-mapping/3838-weighted-word-mapping.java
class Solution {
    public String mapWordWeights(String[] words, int[] weights) {
        StringBuilder result = new StringBuilder();
        for (String word : words) {
            int sumWeight = 0;
            for (char ch : word.toCharArray()) {
                sumWeight += weights[ch - 'a'];
            }
            int value = sumWeight % 26;
            result.append((char)('z' - value));
        }
        return result.toString();
    }
}